### PR TITLE
[Feat]:Home Page and its Functionality

### PR DESCRIPTION
**Pull request by ultronjs**
-Menu UI + Functionality
-Home Page UI
-Add Note Functionality
-Update Note Functionality
-Pinned and UnPinned
-Change the Color of the Notes
-UI for Empty Archive, Label, and Trash Page
There are both commented code +console all over the place.
Commented Code is of API integration that has been commented on now because of the issue in Mockbee.
Console.log is also for the debugging of the same.
-Archive Note API Function
-Archive Note Reducer +Context

### CODE REVIEW

File: src/context/notesContext.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+          type: "Error",
+          msg: "Unable to Fetch Data from API",
+        });
+        console.error(error);
+      }
+    };
+    const addNoteToArchive = async (note) => {
+      console.log(note);
+      notesDispatch({ type: "ADD_NOTE", payload: note });
+      archiveNotesDispatch({ type: "REMOVE_NOTE", payload: note._id });
```
Comment (tusharkandpal) [inline src/context/notesContext.js]: why these two are getting dispatch here ? should be after success response.

File: src/context/notesContext.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        addToast({
+          type: "Error",
+          msg: "Unable to Add Note to Archive",
+        });
+        console.error(error);
+      }
+    };   
+    const restoreNoteFromArchiveNote = async (note) => {
+      archiveNotesDispatch({ type: "DELETE_ARCHIVE_NOTE", payload: note._id });
+      notesDispatch({ type: "ADD_NOTE", payload: note });
```
Comment (tusharkandpal) [inline src/context/notesContext.js]: repeated code in line `144` & `145`

File: src/context/notesContext.js [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        addToast({
+          type: "Error",
+          msg: "Unable to Restore Note from Archive",
+        });
+        console.error(error);
+      }
+
+    }
+    const deleteNoteFromArchiveNote = async (note) => {
+      archiveNotesDispatch({ type: "DELETE_ARCHIVE_NOTE", payload: note });
```
Comment (tusharkandpal) [inline src/context/notesContext.js]: again dispatched in line `165` ?

**Review by tusharkandpal — APPROVED** [review]
suggested few changes. liked your use of toast & its types.

File: src/context/notesContext.js (hunk 1 above)
Comment (ultronjs) [reply src/context/notesContext.js]: Right Now i am getting 500 in API.thatwhy bypassing it right now

File: src/context/notesContext.js (hunk 2 above)
Comment (ultronjs) [reply src/context/notesContext.js]: commented API dispatch on success as API not working right now getting 500

File: src/context/notesContext.js (hunk 3 above)
Comment (ultronjs) [reply src/context/notesContext.js]: sorry forgot to comment that code.
Done.

File: src/components/ArchiveList.jsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+function ArchiveList() {
+  const {
+    archiveNotes,
+    getArchiveNotes,
+    restoreNoteFromArchiveNote,
+    deleteNoteFromArchiveNote,
+  } = useNotes();
+  useEffect(() => {
+    console.log("i m getting triggered");
```
Comment (sushil-mod) [inline src/components/ArchiveList.jsx:15]: remove console log

File: src/components/ColorPallete.jsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+
+function ColorPallete({setNote }) {
```
Comment (sushil-mod) [inline src/components/ColorPallete.jsx:3]: Nice you have made your own color palette

File: src/components/EditNote.jsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  };
+  const changePinnedStatus = () => {
+    setEditNote((prevState) => ({ ...prevState, pinned: !prevState.pinned }));
+  };
+  const changeArchiveStatus = () => {
+    setEditNote((prevState) => ({ ...prevState, archive: !prevState.archive }));
+  };
+  console.log(editNote);
+  // const{children}=parse(content).props
+  // console.log(children)
```
Comment (sushil-mod) [inline src/components/EditNote.jsx:40]: can remove commented code

File: src/components/EditNote.jsx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+          ) : (
+            <BiArchiveIn size={25} onClick={()=>{
+              changeArchiveStatus();
+              addNoteToArchive(editNote)
+            }} />
+          )}
+          <MdLabelOutline size={25} />
+          {/* <MdLabelOutline size={25} />
+        <BsArchive size={25} />
+        <FiTrash size={25} /> */}
```
Comment (sushil-mod) [inline src/components/EditNote.jsx:72]: here also if it is not in use

**Review by sushil-mod — APPROVED** [review]
just a suggestion .overall code is nice

File: src/components/EditNote.jsx (hunk 6 above)
Comment (ultronjs) [reply src/components/EditNote.jsx:40]: Done

File: src/components/ArchiveList.jsx (hunk 4 above)
Comment (ultronjs) [reply src/components/ArchiveList.jsx:15]: Will remove this one in refactoring PR

**Pull request merged**